sky/engine/core/painting/ImageShader.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/engine/core/painting/ImageShader.h"
+
+namespace blink {
+
+PassRefPtr<ImageShader> ImageShader::create() {
+  return adoptRef(new ImageShader());
+}
+
+void ImageShader::initWithImage(CanvasImage* image,
+                                SkShader::TileMode tmx,
+                                SkShader::TileMode tmy,
+                                const Float32List& matrix4,
+                                ExceptionState& es) {
+  ASSERT(image != NULL);
+
+  SkMatrix sk_matrix = toSkMatrix(matrix4, es);
+  if (es.had_exception())
+      return;
+
+  SkBitmap bitmap;
+  image->image()->asLegacyBitmap(&bitmap, SkImage::kRO_LegacyBitmapMode);
+
+  set_shader(adoptRef(SkShader::CreateBitmapShader(bitmap, tmx, tmy, &sk_matrix)));
```

Comment:
Can you add a comment about wishing there was a CreateImageShader that took an SkImage?  Also please file a bug against Skia requesting that feature and link to it from the comment.